src/proof/proof_node_converter.h

```diff
@@ -38,8 +38,19 @@ class ProofNodeManager;
 class ProofNodeConverterCallback
 {
  public:
-  ProofNodeConverterCallback() {}
-  virtual ~ProofNodeConverterCallback() {}
+  ProofNodeConverterCallback();
+  virtual ~ProofNodeConverterCallback();
+  /**
+   * Should proof pn be converted?
+   *
+   * Note this method is called on proof nodes whose premises were not
+   * recursively converted by this callback. We always call convert for such
+   * proof nodes.
```

Comment:
I'm not sure I follow this. Do you mean that it should only be called on pre-visit or something?